Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp

```diff
@@ -925,6 +925,11 @@ void StreamingClient::handleResponseReceived(const ResourceResponse& response)
         return;
     }
 
+    if(response.isRedirected()){
+        g_free(priv->uri);
+        priv->uri = g_strdup(response.url().string().utf8().data());
```

Comment:
Do you need to check if `urlHasSupportedProtocol(url)` like in `webKitWebSrcSetUri`?